### ISSUE TITLE
dweb:/ipfs/$cid not detected on Android

### ISSUE THREAD

**Issue by jefft0** [ipfs-shipyard/ipfs-companion#348]
I have Firefox for Android 57.0.1 and IPFS companion 2.1.0. IPFS-like paths (/ipfs/$cid) are not redirected. (Are they supposed to be in this version?) (The same link redirect correctly in Firefox for Desktop.)

**Comment by lidel** [comment]
Yeah it seems the hack from https://github.com/ipfs-shipyard/ipfs-companion/issues/164#issuecomment-328374052 does not work on Android:
 > ![](https://i.imgur.com/Iw7X2Kbl.png)

Can't do anything about it at this time – Firefox is lacking APIs (https://github.com/ipfs-shipyard/ipfs-companion/issues/164)

**Comment by jefft0** [comment]
OK, thanks. You're aware of the problem. Is there another issue I can track for Android support? If so, you can close this issue.

**Comment by lidel** [comment]
I think for protocol support you can just follow #164. Other things should work on Android. 

If not, please create separate issue for each problem – we don't have robust test suite yet (#145) so such feedback is much appreciated 👍

**Comment by lidel** [comment]
Reopening due to development described in https://github.com/ipfs-shipyard/ipfs-companion/issues/164#issuecomment-355708883

**Comment by cvan** [comment]
Apologies if I'm misunderstanding here. Just wanted to confirm: since https://bugzilla.mozilla.org/show_bug.cgi?id=1428446 has been merged and shipped (woo!), this issue will be moot when PR #359 is merged? Or is there another Firefox (for Android) bug to track here?

**Comment by lidel** [comment]
Yes, this will be closed when PR #359 is merged :)

**Comment by lidel** [comment]
I just smoke-tested latest `master` and Firefox for Android  and it seems to be missing the UI for picking which extension should be a handler for `dweb` protocol. 

Filled an upstream bug:

- https://bugzilla.mozilla.org/show_bug.cgi?id=1445924
